api/src/test/java/org/apache/iceberg/expressions/TestExpressionUtil.java

```diff
@@ -63,6 +64,23 @@ public void testSanitizeIn() {
         ExpressionUtil.toSanitizedString(Expressions.in("test", 34, 345)));
   }
 
+  @Test
+  public void zeroAndNegativeNumberHandling() {
+    Assertions.assertThat(
+            ExpressionUtil.toSanitizedString(
+                Expressions.in(
+                    "test",
+                    0,
+                    Integer.MIN_VALUE,
+                    Integer.MAX_VALUE,
+                    Float.MIN_VALUE,
+                    Float.MAX_VALUE,
+                    Double.MIN_VALUE,
+                    Double.MAX_VALUE)))
+        .isEqualTo(
+            "test IN ((1-digit-int), (10-digit-int), (10-digit-int), (45-digit-float), (39-digit-float), (324-digit-float), (309-digit-float))");
```

Comment:
I'm currently not sure if Float.MIN_VALUE / Double.MIN_VALUE are correctly calculated here